lms/djangoapps/teams/views.py

```diff
@@ -436,8 +436,10 @@ def get(self, request):
                 )
                 return Response(error, status=status.HTTP_400_BAD_REQUEST)
 
-            if course_module.teamsets_by_id[topic_id].is_private_managed:
+            if course_module.teamsets_by_id[topic_id].is_private_managed \
+                    and not has_access(request.user, 'staff', course_key):
                 result_filter.update({'membership__user__username': request.user})
```

Comment:
From a DRY standpoint, probably want to look into leveraging our API functions for this. e.g. `has_specific_team_access` allows staff to view a team.

Also, the `\` isn't technically needed.

Reply:
In has_specific_team_access would not be appropriate, as according to its doc string:
To have access to a team a user must:
        - Be course staff
        OR
        - be in the correct bubble
        - be in the team if it is private